core/src/bundle_stage.rs

```diff
@@ -322,6 +313,19 @@ impl BundleStage {
             ),
         );
```

Comment:
early return needs to occur after metrics accumulation

Reply:
i wonder if we need to move the qos_service.report_metrics too?

Reply:
yes, good catch